### PR TITLE
Extension of authentication methods

### PR DESCRIPTION
**Pull request by brtkwr**

### CODE REVIEW

File: library/os_container_infra.py [hunk 1]
```diff
@@ -94,13 +97,21 @@ def __init__(self, **kwargs):
...
-        self.cloud = openstack.connect(cloud=cloud)
+    def connect(self, **kwargs):
+        if kwargs['auth_type'] == 'environment':
+            self.cloud = openstack.connect()
+        elif kwargs['auth_type'] == 'cloud':
+            self.cloud = openstack.connect(cloud=kwargs['cloud'])
+        elif kwargs['auth_type'] == 'password':
+            self.cloud = openstack.connect(**kwargs['auth'])
+        else:
+            raise OpenStackAuthConfig
```
Comment (markgoddard) [inline library/os_container_infra.py]: Does this provide a meaningful error to the user?

File: library/os_container_infra.py [hunk 2]
```diff
@@ -122,7 +133,8 @@ def apply(self):
                 status = self.result['status']
                 # If the cluster is in a failed status, raise error:
                 if status.endswith('FAILED'):
-                    raise OpenStackError(self.result['faults'])
+                    raise OpenStackError(status)
```
Comment (markgoddard) [inline library/os_container_infra.py]: Showing the faults seemed more helpful...

**Comment by markgoddard** [comment]
Has merge conflicts

**Comment by brtkwr** [comment]
Resolved.

File: library/os_container_infra.py (hunk 1 above)
Comment (brtkwr) [reply library/os_container_infra.py]: fixed

File: library/os_container_infra.py (hunk 2 above)
Comment (brtkwr) [reply library/os_container_infra.py]: fixed

**Comment by brtkwr** [comment]
Addressed your comments @markgoddard, ready for another review.

**Review by markgoddard — APPROVED** [review]

**Pull request merged**